.github/workflows/style-checks.yml

```diff
@@ -0,0 +1,27 @@
+name: Black  # TODO: add isort and flake8 later
+
+on: 
+  pull_request: {}
+  push:
+    branches: master
+    tags: "*"
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Setup Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.10'
+
+      - name: Install dependencies with pip
+        run: |
+          pip install --upgrade pip wheel
+          pip install .[test]
+
+      # - run: isort --check-only .
+      - run: black --check .
```

Comment:
Since you're using pre-commit, may I recommend the following action instead:

```
      - uses: pre-commit/action@v3.0.0
```

Reply:
This is pretty cool. I did not know this existed. Personally I'm a little torn on using it because there's not much gained by using it and it leaves us dependent on an externally managed action. There's also a slight difference in what is being run in the precommit hooks right now. For the sake of speed, they only run black on what is being committed, while the current GHA is running black on everything.
If others prefer using this action, though, I don't feel super strongly about it.

Reply:
I'm not a dev, so please remember that I cannot speak for the project at all :-). I understand what you're saying about external actions, although it is pretty official. Perhaps at least run `pre-commit run -a` instead of black directly? It reduces the chance that at some point someone might want to change a parameter or something and will need to remember to sync in both places.

IIRC, I think the action it can also be configured to add comments with places that were missed if you pass a token (but possibly not on a PR from a fork). It also includes a `git diff` after it runs, so instead of just a binary "failed", you can see what it fixed, which can be helpful for new contributors who aren't aware of the coding conventions.